src/meta/src/manager/user.rs

```diff
@@ -39,38 +41,53 @@ pub struct UserManager<S: MetaStore> {
 }
 
 pub struct UserManagerInner {
-    user_info: HashMap<UserName, UserInfo>,
-    user_grant_relation: HashMap<UserName, HashSet<UserName>>,
+    user_info: HashMap<UserId, UserInfo>,
+    user_grant_relation: HashMap<UserId, HashSet<UserId>>,
+    all_users: HashSet<String>,
 }
 
-fn get_relation(user_info: &HashMap<UserName, UserInfo>) -> HashMap<UserName, HashSet<UserName>> {
-    let mut user_grant_relation: HashMap<UserName, HashSet<UserName>> = HashMap::new();
-    for (user_name, info) in user_info {
+fn get_relation(user_info: &HashMap<UserId, UserInfo>) -> HashMap<UserId, HashSet<UserId>> {
+    let mut user_grant_relation: HashMap<UserId, HashSet<UserId>> = HashMap::new();
+    for (user_id, info) in user_info {
         for grant_privilege_item in &info.grant_privileges {
             for option in &grant_privilege_item.action_with_opts {
                 user_grant_relation
-                    .entry(option.get_granted_by().to_string())
+                    .entry(option.get_granted_by())
                     .or_insert_with(HashSet::new);
                 let relation_item = user_grant_relation
-                    .get_mut(option.get_granted_by())
+                    .get_mut(&option.get_granted_by())
                     .unwrap();
-                relation_item.insert(user_name.clone());
+                relation_item.insert(*user_id);
             }
         }
     }
     user_grant_relation
 }
 
 impl UserManagerInner {
-    pub fn new(user_info: HashMap<UserName, UserInfo>) -> Self {
+    pub fn new(users: Vec<UserInfo>) -> Self {
+        let all_users = HashSet::from_iter(users.iter().map(|user| user.name.clone()));
+        let user_info = HashMap::from_iter(users.into_iter().map(|user| (user.id, user)));
         let user_grant_relation = get_relation(&user_info);
         Self {
             user_info,
             user_grant_relation,
+            all_users,
         }
     }
 
-    pub fn get_user_info(&self) -> &HashMap<UserName, UserInfo> {
+    fn create_user(&mut self, user: UserInfo) {
+        self.all_users.insert(user.name.clone());
+        self.user_info.insert(user.id, user);
+    }
+
+    fn drop_user(&mut self, user_id: UserId) {
+        if let Some(user) = self.user_info.remove(&user_id) {
+            self.all_users.remove(&user.name);
+        }
```

Comment:
Add some comments that user in `user_grant_relation` (as key or value) are already checked before entering this function.